casper/src/main/scala/coop/rchain/casper/util/rholang/RuntimeManager.scala

```diff
@@ -174,69 +249,130 @@ class RuntimeManagerImpl[F[_]: Concurrent] private[rholang] (
         Bond(validatorName, stakeAmount)
     }.toList
 
-  def getData(hash: ByteString, channel: Par): F[Seq[Par]] =
-    withResetRuntime(hash, runtime => {
-      runtime.space.getData(channel).map(_.flatMap(_.a.pars))
-    })
+  def getData(start: StateHash)(name: Par): F[Seq[Par]] =
+    withResetRuntime(start)(getData(_)(name))
+
+  private def getData(
+      runtime: Runtime[F]
+  )(name: Par): F[Seq[Par]] =
+    runtime.space.getData(name).map(_.flatMap(_.a.pars))
+
+  private def getResult(
+      runtime: Runtime[F]
+  )(name: String = "__SCALA__"): F[Seq[Par]] =
+    runtime.space.getData(Par().withExprs(Seq(Expr(GString(name))))).map(_.flatMap(_.a.pars)) <*
+      computeEffect(runtime)(ConstructDeploy.sourceDeployNow(s"""for(_<-@"$name"){ Nil }"""))
 
   def getContinuation(
-      hash: ByteString,
-      channels: immutable.Seq[Par]
-  ): F[Seq[(Seq[BindPattern], Par)]] =
-    withResetRuntime(
-      hash,
-      runtime => {
-        runtime.space
-          .getWaitingContinuations(channels)
-          .map(
-            results =>
-              for {
-                result <- results.filter(_.continuation.taggedCont.isParBody)
-              } yield (result.patterns, result.continuation.taggedCont.parBody.get.body)
-          )
-      }
+      start: StateHash
+  )(channels: Seq[Par]): F[Seq[(Seq[BindPattern], Par)]] =
+    withResetRuntime(start)(
+      _.space
+        .getWaitingContinuations(channels)
+        .map(
+          _.filter(_.continuation.taggedCont.isParBody)
+            .map(result => (result.patterns, result.continuation.taggedCont.parBody.get.body))
+        )
     )
 
   private def newEval(
```

Comment:
The commit name `Eliminate superfluous reset from newEval` is a bit misleading. "Superfluous" would suggest it being removed whatsoever, while it's just been moved up to `computeState`.

Also, separating the changes to `replayComputeState` would prevent me bing mislead about its behavior (not) being changed in this commit.

Reply:
I'm not sure we're talking about the same commit. I also think we can leave the commit I mentioned above as is (unless I made other comments to it).

Trying to answer your question: the changes to `replayComputeState` in that commit would best be split as a separate commit saying sth like 'rename the `hash` parameter to `start` in `replayComputeState`', because of the reason I gave above (I thought the reset was also eliminated from replay for quite some time, partly because of the comment you added too - which could easily be yet another commit.)

All this conversation is only useful if it makes our future PR reviews more understandable - no need for changes now :)